packages/web3-utils/src/converters.ts

```diff
@@ -0,0 +1,340 @@
+		}
+
+		if ((typeof value === 'number' || typeof value === 'bigint') && BigInt(value) < BigInt(0)) {
+			return `-0x${value.toString(16).substr(1)}`;
+		}
+
+		if (
+			(typeof value === 'number' || typeof value === 'bigint') &&
+			BigInt(value) >= BigInt(0)
+		) {
+			return `0x${value.toString(16)}`;
+		}
+
+		if (typeof value === 'string' && isHexStrict(value)) {
+			return numberToHex(hexToNumber(value));
+		}
+
+		if (typeof value === 'string' && !isHexStrict(value)) {
+			return numberToHex(BigInt(value));
+		}
+	} catch {
+		throw new Error(errorMessage(value, 'not a valid integer'));
+	}
+
+	throw new Error(errorMessage(value, 'not a valid integer'));
+};
+/**
+ * Converts value to it's hex representation @alias `numberToHex`
+ */
+export const fromDecimal = numberToHex;
+
+/**
+ * Converts value to it's decimal representation in string
+ */
+export const hexToNumberString = (data: HexString): string => hexToNumber(data).toString();
+
+/**
+ * Should be called to get hex representation (prefixed by 0x) of utf8 string
+ */
+export const utf8ToHex = (str: string): HexString => {
+	if (typeof str !== 'string') {
+		throw new Error(errorMessage(str, 'not a valid string'));
+	}
+
+	// To be compatible with 1.x trim null character
+	// eslint-disable-next-line no-control-regex
+	let strWithoutNullCharacter = str.replace(/^(?:\u0000)/, '');
+	// eslint-disable-next-line no-control-regex
+	strWithoutNullCharacter = strWithoutNullCharacter.replace(/(?:\u0000)$/, '');
+
+	return `0x${Buffer.from(strWithoutNullCharacter, 'utf8').toString('hex')}`;
+};
+
+/**
+ * @alias `utf8ToHex`
+ */
+
+export const fromUtf8 = utf8ToHex;
+/**
+ * @alias `utf8ToHex`
+ */
+export const stringToHex = utf8ToHex;
+
+/**
+ * Should be called to get utf8 from it's hex representation
+ */
+export const hexToUtf8 = (str: HexString): string => bytesToBuffer(str).toString('utf8');
+
+/**
+ * @alias `hexToUtf8`
+ */
+export const toUtf8 = hexToUtf8;
+
+/**
+ * @alias `hexToUtf8`
+ */
+export const hexToString = hexToUtf8;
+
+/**
+ * Should be called to get hex representation (prefixed by 0x) of ascii string
+ */
+export const asciiToHex = (str: string): HexString => {
+	if (typeof str !== 'string') {
+		throw new Error(errorMessage(str, 'not a valid string'));
+	}
+
+	return `0x${Buffer.from(str, 'ascii').toString('hex')}`;
+};
+
+/**
+ * @alias `asciiToHex`
+ */
+export const fromAscii = utf8ToHex;
+
+/**
+ * Should be called to get ascii from it's hex representation
+ */
+export const hexToAscii = (str: HexString): string => bytesToBuffer(str).toString('ascii');
+
+/**
+ * @alias `hexToAscii`
+ */
+export const toAscii = hexToUtf8;
+
+/**
+ * Auto converts any given value into it's hex representation.
+ */
+export const toHex = (
+	value: Numbers | Bytes | Address | boolean,
+	returnType?: boolean,
+): HexString | ValueTypes => {
+	if (typeof value === 'string' && isAddress(value)) {
+		return returnType ? 'address' : `0x${value.toLowerCase().replace(/^0x/i, '')}`;
```

Comment:
`.replace(/^0x/i, '')` could probably get removed after `isAddress` is implemented

Reply:
Not sure, I copied it exactly from old implementation.